### DISCUSSION TITLE
Filament and Pressure advance management

### DISCUSSION THREAD

**Discussion by ajshell1** [jschuh/klipper-macros#105] [Ideas]
I'd like to preface this by saying that I'm a big fan of this project so far. This is why I want to see it get even better.

With this in mind, I think it would be really cool if it could integrate some kind of system for keeping track of the types of filaments we're using and their properties.

The most basic way to manage pressure advance is to change the printer.cfg file every time you change your filament.

My current system uses a macro in PrusaSlicer's filament custom gcode. That is, my PETG profile calls the FILAMENT_PETG macro, which sets my pressure advance to a specific value.

I'm curious to see if this project could come up with something more elegant.

**Comment by jschuh** [comment]
I guess my opinion is that the slicer filament profiles are already the best solution, because you know when slicing the file what filament you're going to use, whereas the printer can't know unless the slicer tells it.

I did actually play around with something that could use the default Prusa Slicer profiles when I wrote [the M900 macro.](https://github.com/jschuh/klipper-macros/blob/a7a34ab1604643a591ed4af62fb7188c800a781e/velocity.cfg#LL67C1-L80C1)

I even spent a bit of time investigating how to accurately remap Marlin's linear advance K factors to Klipper's pressure advance values. But I didn't see an obviously good solution and it all just seemed like it was more trouble than it was worth.
